File: module-3-java/spring-boot-tut/src/main/java/com/example/demo/model/UserRFM.java
package com.example.demo.model;

import lombok.AllArgsConstructor;
import lombok.Data;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;
import java.util.Date;

/**
 * The type User rfm.
 */
@Entity
@Data
@Table(name = "user_rfm")
public class UserRFM {

    @Id
    private int id;

    @Column(name = "user_id")
    private String userId;

    @Column(name = "recorded_date")
    private Date recordedDate;

    @Column(name = "recency")
    private int recency;

    @Column(name = "frequency")
    private int frequency;

    @Column(name = "monetary")
    private double monetary;

    @Column(name = "r_score")
    private int rScore;

    @Column(name = "f_score")
    private int fScore;

    @Column(name = "m_score")
    private int mScore;

    @Column(name = "rfm_score")
    private String rfmScore;
}

File: module-3-java/bi-tool/pages/components/userRFM.js
import MUIDataTable from 'mui-datatables'

const columns = [
	{
		name: 'recency',
		label: 'Recency'
	},
	{
		name: 'frequency',
		label: 'Frequency'
	},
	{
		name: 'monetary',
		label: 'Monetary'
	},
	{
		name: 'rscore',
		label: 'R Score'
	},
	{
		name: 'fscore',
		label: 'F Score'
	},
	{
		name: 'mscore',
		label: 'M Score'
	},
	{
		name: 'rfmScore',
		label: 'RFM Score'
	}
]

const options = {
	download: false,
	print: false
}
const UserRFM = ({ data }) => {
	return (
		<MUIDataTable title={'User RFM'} data={[ data ]} columns={columns} options={options} className='w-2/3 my-2' />
	)
}

export default UserRFM
